src/main/java/com/google/api/generator/gapic/composer/ServiceClientSampleCodeComposer.java

```diff
@@ -0,0 +1,121 @@
+      Service service, Map<String, TypeNode> types) {
+    String settingsVarName = JavaStyle.toLowerCamelCase(getSettingsName(service.name()));
+    TypeNode settingsVarType = types.get(getSettingsName(service.name()));
+    VariableExpr settingsVarExpr = createVariableExpr(settingsVarName, settingsVarType);
+    MethodInvocationExpr newBuilderMethodExpr =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(settingsVarType)
+            .setMethodName("newBuilder")
+            .build();
+    TypeNode fixedCredentialProvideType =
+        TypeNode.withReference(ConcreteReference.withClazz(FixedCredentialsProvider.class));
+    MethodInvocationExpr credentialArgExpr =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(fixedCredentialProvideType)
+            .setArguments(ValueExpr.withValue(StringObjectValue.withValue("myCredentials")))
+            .setMethodName("create")
+            .build();
+    MethodInvocationExpr credentialsMethodExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(newBuilderMethodExpr)
+            .setArguments(credentialArgExpr)
+            .setMethodName("setCredentialsProvider")
+            .build();
+    MethodInvocationExpr buildMethodExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(credentialsMethodExpr)
+            .setReturnType(settingsVarType)
+            .setMethodName("build")
+            .build();
+    Expr initSettingsVarExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(settingsVarExpr.toBuilder().setIsDecl(true).build())
+            .setValueExpr(buildMethodExpr)
+            .build();
+
+    String clientName = JavaStyle.toLowerCamelCase(getClientClassName(service.name()));
+    TypeNode clientType = types.get(getClientClassName(service.name()));
+    VariableExpr clientVarExpr = createVariableExpr(clientName, clientType);
+    MethodInvocationExpr createMethodExpr =
+        MethodInvocationExpr.builder()
+            .setStaticReferenceType(clientType)
+            .setArguments(settingsVarExpr)
+            .setMethodName("create")
+            .setReturnType(clientType)
+            .build();
+    Expr initClientVarExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(clientVarExpr.toBuilder().setIsDecl(true).build())
+            .setValueExpr(createMethodExpr)
+            .build();
+
+    return writeSampleCode(Arrays.asList(initSettingsVarExpr, initClientVarExpr));
+  }
+
+  // ======================================== Helpers ==========================================//
+
+  private static String getClientClassName(String serviceName) {
+    return String.format(CLASS_NAME_PATTERN, serviceName);
+  }
+
+  private static String getSettingsName(String serviceName) {
+    return String.format(SETTINGS_NAME_PATTERN, serviceName);
+  }
+
+  private static String writeSampleCode(List<Expr> exprs) {
```

Comment:
What about centralizing this into a SampleCodeCreator or SampleCodeWriter in `samplecode`?

Reply:
Greate idea. I add one in #499

Reply:
Add TODO to refactor once #499 merged.